js/main.js

```diff
@@ -11,20 +11,42 @@ function urlHandler(event) {
 photoUrl.addEventListener('input', urlHandler);
 
 var submitForm = document.querySelector('#entry-form');
-
 function submitHandler(event) {
   event.preventDefault();
   var inputs = submitForm.elements;
   var titleValue = inputs.title.value;
   var urlValue = inputs.url.value;
   var notesValue = inputs.notes.value;
-  var entryValues = {
-    titleValue,
-    urlValue,
-    notesValue
-  };
-  data.nextEntryId++;
-  data.entries.unshift(entryValues);
+
+  if (data.editing === null) {
+    var entryValues = {
+      titleValue,
+      urlValue,
+      notesValue,
+      entryId: data.nextEntryId
+    };
+    data.entries.unshift(entryValues);
+    data.nextEntryId++;
+  }
+
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries[i].entryId === data.editing) {
+      entryValues = {
+        titleValue,
+        urlValue,
+        notesValue,
+        entryId: data.nextEntryId - 1
```

Comment:
This is incorrect, it would only give you the `id` for the last entry created. You need this to have the same id it has always had, which is stored in the object for this entry (`data.entries[i]`)